packages/core/src/xr/session/XRSessionManager.ts

```diff
@@ -0,0 +1,171 @@
+import { IHardwareRenderer, IXRFeature, IXRSession } from "@galacean/engine-design";
+import { Engine } from "../../Engine";
+import { XRSessionMode } from "./XRSessionMode";
+import { XRSessionState } from "./XRSessionState";
+
+/**
+ * XRSessionManager manages the life cycle of XR sessions.
+ */
+export class XRSessionManager {
+  protected _session: IXRSession;
+  protected _state: XRSessionState = XRSessionState.None;
+
+  private _rhi: IHardwareRenderer;
+  private _raf: (callback: FrameRequestCallback) => number;
+  private _caf: (id: number) => void;
+
+  /**
+   * Return the current session state.
+   */
+  get state(): XRSessionState {
+    return this._state;
+  }
+
+  /**
+   * Return the current session.
+   */
+  get session(): IXRSession {
+    return this._session;
+  }
+
+  /**
+   * Return a list of supported frame rates.(only available in-session)
+   */
+  get supportedFrameRate(): Float32Array {
+    return this._session.supportedFrameRates;
+  }
+
+  /**
+   * Return the current frame rate as reported by the device.
+   */
+  get frameRate(): number {
+    return this._session.frameRate;
+  }
+
+  /**
+   * Returns requestAnimationFrame in XR.
+   */
+  get requestAnimationFrame(): (callback: FrameRequestCallback) => number {
+    if (this._state === XRSessionState.Running) {
+      return this._session.requestAnimationFrame;
+    } else {
+      return this._raf;
+    }
+  }
+
+  /**
+   * Returns cancelAnimationFrame in XR.
+   */
+  get cancelAnimationFrame(): (id: number) => void {
```

Comment:
`requestAnimationFrame` and `cancelAnimationFrame` should internal? Developer should call this?